package foo

fun foo() {
    <!INVISIBLE_MEMBER, FUNCTION_CALL_EXPECTED, VARIABLE_EXPECTED!>System<!> = <!ERROR_COMPILE_TIME_VALUE!>5<!>
    <!INVISIBLE_MEMBER, FUNCTION_CALL_EXPECTED!>System<!><!UNNECESSARY_NOT_NULL_ASSERTION!>!!<!>
    if (<!INVISIBLE_MEMBER, FUNCTION_CALL_EXPECTED, TYPE_MISMATCH_IN_CONDITION!>System<!>)
       <!INVISIBLE_MEMBER, FUNCTION_CALL_EXPECTED!>System<!><!UNRESOLVED_REFERENCE!>++<!>
    java.<!EXPRESSION_EXPECTED_NAMESPACE_FOUND!>util<!><!INC_DEC_SHOULD_NOT_RETURN_UNIT, DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!>++<!>
    System.out.println("This is ok")
}